docs/tutorials/ssr/README.md

```diff
@@ -1,84 +1,135 @@
 # Firebase Auth in SSR/Universal Mode
 
 <Badge text="EXPERIMENTAL" type="warn"/>
+
 This module provides an option for the easy setup of **server-side authentication** as described in [this article](https://firebase.google.com/docs/auth/web/service-worker-sessions) of the official Documentation.
 
-### Step 0 - Install Dependencies
+## Step 0 - Install Dependencies
 
-In addition to @nuxtjs/firebase, install these two dependencies:
+Install `firebase-admin` and `@nuxtjs/pwa`:
 
 ```bash
-yarn add firebase-admin # OR npm i firebase-admin
-yarn add jwt-decode # OR npm i jwt-decode
+yarn add firebase-admin @nuxtjs/pwa # OR npm i firebase-admin @nuxtjs/pwa
 ```
 
-### Step 1 - Initialize Firebase Auth
+## Step 1 - Enable SSR functionality and configure workbox to include the auth service worker
+
+Use the [auth.ssr option](/guide/options/#ssr).
 
-Use the [auth.initialize option](/guide/options/#auth) with at least `onSuccessAction` defined and set `ssr = true`. Make sure to create the respective action & mutation to save the authUser to the state.
+In `nuxt.config.js`:
 
 ```js
-// In the module config in nuxt.config.js:
-auth: {
-  initialize: {
-    onSuccessAction: 'handleSuccessfulAuthentication',
-    ssr: true
+module.exports = {
+  // ...
+  modules: [
+    // ...
+    '@nuxtjs/pwa',
+    '@nuxtjs/firebase'
+  ],
+  firebase {
+    // ...
+    services: {
+      auth: {
+        ssr: true
+      }
+      // ,,,
+    }
+  },
+  pwa: {
+    workbox: {
+      importScripts: [
+        // ...
+        '/firebase-auth-sw.js'
+      ],
+      // by default the workbox module will not install the service worker in dev environment to avoid conflicts with HMR
+      // only set this true for testing and remember to always clear your browser cache in development
+      dev: false
+    }
   }
 },
 ```
 
-Set up your action like so to (1) register the service worker and (2) commit the authUser:
+## Step 2 - Setup Mutations and/or Actions to handle User authentication
+
+If you don't rely on a full `firebase.User` object you can reuse the client action/mutation configured for [`auth.initialize`](/guide/options/#initialize).  
+See [Step 3](#step-3-retrieve-the-server-user) to verify if required properties are present.
+
+Example action:
 
 ```js
-handleSuccessfulAuthentication({ commit }, { authUser }) {
-  // Install servicerWorker if supported on sign-in/sign-up page.
-  if (process.browser && 'serviceWorker' in navigator) {
-    navigator.serviceWorker.register('/firebase-auth-sw.js', { scope: '/' })
+async onAuthStateChangedAction({ commit, dispatch }, { authUser, claims }) {
+  if (!authUser) {
+    await dispatch('cleanupAction')
+
+    return
   }
-  commit('SET_AUTH_USER', { authUser })
-},
+
+  // use custom claims (see https://firebase.google.com/docs/auth/admin/custom-claims)
+  if (claims.admin) {
```

Comment:
Two points:

- Custom claims don't need to be used necessarily in this action, it's just an example what one could do. The action can also be used to do other asynchronous calls not related to custom claims at all.

- The "admin" custom claim is not a given property and may not even exist / be used in many projects.

Due to this I would prefer to **not** use it as an example everywhere in the docs and just write something like `// do other async calls`, just for not confusing people with this.

(also in the other examples further up)

Reply:
I have renamed the requested property to `claims.custom_claim` and added a reference to the [corresponding firebase guide](https://firebase.google.com/docs/auth/admin/custom-claims).

I feel the tutorial should provide reference material where possible.

Let me know what you think

Reply:
All good - thanks. 👍